src/System.Windows.Forms/src/System/Windows/Forms/CheckBox.cs

```diff
@@ -546,10 +553,18 @@ protected override void OnClick(EventArgs e)
                         else
                         {
                             CheckState = CheckState.Unchecked;
+                            AccessibilityObject.RaiseAutomationPropertyChangedEvent(
+                            UiaCore.UIA.ToggleToggleStatePropertyId,
+                            UiaCore.ToggleState.On,
+                            UiaCore.ToggleState.Off);
                         }
                         break;
                     default:
                         CheckState = CheckState.Unchecked;
+                        AccessibilityObject.RaiseAutomationPropertyChangedEvent(
+                            UiaCore.UIA.ToggleToggleStatePropertyId,
+                            UiaCore.ToggleState.On,
+                            UiaCore.ToggleState.Off);
```

Comment:
Please add the test for `IsHandleCreated`. Please also consider to move it to a private method like:
```
RaiseUiaToggleStateChangedEvent(bool toggleStateOn)
{
    if (owner.IsHandleCreated)
    {
        AccessibilityObject.RaiseAutomationPropertyChangedEvent(
            UiaCore.UIA.ToggleToggleStatePropertyId,
            toggleStateOn ? UiaCore.ToggleState.On : UiaCore.ToggleState.Off,
            toggleStateOn ? UiaCore.ToggleState.Off : UiaCore.ToggleState.On);
    }
}
```